### PR TITLE
feat: adds some tables to information_schema

### PR DESCRIPTION
**Pull request by killme2008**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

Part of #2931 

Main changes:

* Adds `engines`, `column_privileges` and `column_statistics` to `information_schema`.
* Impl some helper functions and macros to add tables for `information_schema`.

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [x]  I have added the necessary unit tests and integration tests.

## Refer to a related PR or issue link (optional)

#2931

### CODE REVIEW

File: src/catalog/src/information_schema.rs [hunk 1]
```diff
@@ -32,14 +36,38 @@ use table::metadata::{
...
-pub const TABLES: &str = "tables";
-pub const COLUMNS: &str = "columns";
+lazy_static! {
+    // Memory tables in `information_schema`.
+    static ref MEMORY_TABLES: Vec<&'static str> = vec![
+        ENGINES,
+        COLUMN_PRIVILEGES,
+        COLUMN_STATISTICS
+    ];
+}
```
Comment (waynexia) [inline src/catalog/src/information_schema.rs:54]:
```suggestion
/// Memory tables in `information_schema`.
const MEMORY_TABLES: &[&str; 3] = &[ENGINES, COLUMN_PRIVILEGES, COLUMN_STATISTICS];
```

Comment (killme2008) [reply src/catalog/src/information_schema.rs:54]: We will continue adding tables, it's cumbersome to modify the array length every time.

Comment (waynexia) [reply src/catalog/src/information_schema.rs:54]: Ah, the length is not required. Try this:

```suggestion
/// Memory tables in `information_schema`.
const MEMORY_TABLES: &[&str] = &[ENGINES, COLUMN_PRIVILEGES, COLUMN_STATISTICS];
```

**Comment by shuiyisong** [comment]
https://github.com/GreptimeTeam/greptimedb/blob/9ce9421850dd8d57cc4940a5d3f03785cc9df811/src/catalog/src/kvbackend/manager.rs#L241

We might need to add new memtable names to `table_names()` and `table_exist()` in `

**Review by sunng87 — APPROVED** [review]

**Comment by killme2008** [comment]
> https://github.com/GreptimeTeam/greptimedb/blob/9ce9421850dd8d57cc4940a5d3f03785cc9df811/src/catalog/src/kvbackend/manager.rs#L241
> 
> We might need to add new memory table names to `table_names()` and `table_exist()` in `SystemCatalog` so that they are displayed on `show tables`

OMG, good catch! Let me find some way to make it more readable.

File: src/catalog/src/information_schema.rs [hunk 2]
```diff
@@ -80,7 +114,8 @@ impl InformationSchemaProvider {
     }
 
     fn information_table(&self, name: &str) -> Option<InformationTableRef> {
-        match name.to_ascii_lowercase().as_str() {
+        let name = name.to_ascii_lowercase();
+        match name.clone().as_str() {
```
Comment (zhongzc) [inline src/catalog/src/information_schema.rs:118]:
```suggestion
        match name.to_ascii_lowercase().as_str() {
```

**Review by zhongzc — APPROVED** [review]

**Comment by killme2008** [comment]
@waynexia @shuiyisong PTAL

**Pull request merged**